browser/brave_ads/request_ads_enabled_api_browsertest.cc

```diff
@@ -0,0 +1,293 @@
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    https_server_->ServeFilesFromDirectory(test_data_dir);
+
+    ASSERT_TRUE(https_server_->Start());
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    // HTTPS server only serves a valid cert for localhost, so this is needed
+    // to load pages from other hosts without an error.
+    command_line->AppendSwitch(switches::kIgnoreCertificateErrors);
+  }
+
+  content::WebContents* OpenRequestAdsEnabledPopup(
+      content::WebContents* contents) {
+    content::WebContents* popup_contents = nullptr;
+    auto check_load_is_rewards_panel =
+        [](content::WebContents** popup_contents,
+           const content::NotificationSource& source,
+           const content::NotificationDetails&) -> bool {
+      auto web_contents_source =
+          static_cast<const content::Source<content::WebContents>&>(source);
+      *popup_contents = web_contents_source.ptr();
+
+      // Check that this notification is for the Rewards panel and not, say,
+      // the extension background page.
+      std::string url = (*popup_contents)->GetLastCommittedURL().spec();
+      std::string rewards_panel_url = std::string("chrome-extension://") +
+                                      brave_rewards_extension_id +
+                                      "/request_ads_enabled_panel.html";
+      return url == rewards_panel_url;
+    };
+
+    content::WindowedNotificationObserver popup_observer(
+        content::NOTIFICATION_LOAD_COMPLETED_MAIN_FRAME,
+        base::BindRepeating(check_load_is_rewards_panel, &popup_contents));
+
+    EXPECT_TRUE(ExecuteScript(contents, kBraveRequestAdsEnabled));
+
+    // Wait for the popup to load
+    popup_observer.Wait();
+    rewards_browsertest_util::WaitForElementToAppear(
+        popup_contents, "[data-test-id='rewards-onboarding-main-button']");
+
+    return popup_contents;
+  }
+
+  net::EmbeddedTestServer* https_server() { return https_server_.get(); }
+
+ protected:
+  base::test::ScopedFeatureList feature_list_;
+
+ private:
+  std::unique_ptr<net::EmbeddedTestServer> https_server_;
+};
+
+class RequestAdsEnabledApiTestEnabled : public RequestAdsEnabledApiTestBase {
+ public:
+  RequestAdsEnabledApiTestEnabled() {
+    feature_list_.InitAndEnableFeature(
+        brave_ads::features::kRequestAdsEnabledApi);
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(RequestAdsEnabledApiTestEnabled, AdsAlreadyEnabled) {
+  brave_ads::AdsService* ads_service =
+      brave_ads::AdsServiceFactory::GetForProfile(browser()->profile());
+  ads_service->SetEnabled(true);
+
+  GURL url = https_server()->GetURL(kAllowedDomain, "/simple.html");
+
+  ui_test_utils::NavigateToURL(browser(), url);
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  WaitForLoadStop(contents);
+  EXPECT_EQ(url, contents->GetURL());
+  bool has_api;
+  EXPECT_TRUE(ExecuteScriptAndExtractBool(
```

Comment:
`ExecuteScript` functions are deprecated in favor of `EvalJs`

Reply:
Fixed